src/schema/firefox-schemas-import.js

```diff
@@ -68,7 +68,18 @@ export function rewriteOptionalToRequired(schema) {
     const value = schema[key];
     if (!Array.isArray(value) && typeof value === 'object') {
       const { optional, ...rest } = value;
-      if (!optional) {
+      if (Array.isArray(rest.allOf)) {
```

Comment:
This is a bit of an odd to read chunk of code. I get it after looking at the issue, but I wouldn't mind a link back to the issue just to explain the reasoning behind it.